Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeDrawingAreaProxyMac.mm

```diff
@@ -206,20 +208,38 @@ explicit RemoteLayerTreeDisplayLinkClient(WebPageProxyIdentifier pageID)
 
 void RemoteLayerTreeDrawingAreaProxyMac::commitTransientZoom(double scale, FloatPoint origin)
 {
-    LOG_WITH_STREAM(ViewGestures, stream << "RemoteLayerTreeDrawingAreaProxyMac::commitTransientZoom - scale " << scale << " origin " << origin);
-
+    auto rootNode = downcast<ScrollingTreeFrameScrollingNode>(m_webPageProxy.scrollingCoordinatorProxy()->rootNode());
+    auto scrollPosition = rootNode->currentScrollPosition();
+    auto visibleContentRect = rootNode->layoutViewport();
+    visibleContentRect.setX(scrollPosition.x());
+    visibleContentRect.setY(scrollPosition.y());
+    
+    FloatPoint constrainedOrigin = visibleContentRect.location();
+    constrainedOrigin.moveBy(-origin);
+
+    IntSize scaledTotalContentsSize = roundedIntSize(rootNode->totalContentsSize());
+    scaledTotalContentsSize.scale(scale / rootNode->frameScaleFactor());
```

Comment:
Is `rootNode->frameScaleFactor()` the last scale that we committed to the web process? Is that what we want?

Reply:
Yes we want to scale this by the difference in scale, based on how TiledCoreAnimationDrawingArea uses `scale / m_webPage.totalScaleFactor()`.